packages/core/src/lib/TwaGenerator.ts

```diff
@@ -224,54 +225,67 @@ export class TwaGenerator {
    * @param {String} targetDirectory the directory where the project will be created
    * @param {Object} twaManifest configurations values for the project.
    */
-  async createTwaProject(targetDirectory: string, twaManifest: TwaManifest): Promise<void> {
+  async createTwaProject(targetDirectory: string, twaManifest: TwaManifest,
+      reportProgress: twaGeneratorProgress = noOpProgress): Promise<void> {
+    let currentProgress = 0;
+    const totalProgress = 8;
```

Comment:
This worries me a bit. I can't see a way around it though unless we break each of these chunks of code into a "step" and then have a list of steps which we iterate through and perform. I'm not sure that's worth it though.

Could we at least add a test to check we finish with `currentProgress == totalProgress`?

Reply:
Added a wrapper class for progress (may be overkill?). It will ensure a) trying to update progress after reaching total throws an error and b) the last progress call should be `done()` and if current and total don't match, it throws an Error.

I haven't made the class public, but if we need something similar elsewhere, it may be worth moving it somewhere else and making it public.

Reply:
PS: I didn't add tests, as we need a larger refactor of `TwaGenerator` to make it testable.